internal/report/source.go

```diff
@@ -483,6 +514,55 @@ func getMissingFunctionSource(filename string, asm map[int][]assemblyInstruction
 	return fnodes, filename
 }
 
+// sourceReader provides access to source code with caching of file contents.
+type sourceReader struct {
+	searchPath string
+
+	// files maps from path name to a list of lines.
+	// files[*][0] is unused since line numbering starts at 1.
+	files map[string][]string
+
+	errors map[string]error
+}
+
+func newSourceReader(searchPath string) *sourceReader {
+	return &sourceReader{
+		searchPath,
+		make(map[string][]string),
+		make(map[string]error),
+	}
+}
+
+func (reader *sourceReader) fileError(path string) error {
+	return reader.errors[path]
+}
+
+func (reader *sourceReader) line(path string, lineno int) (string, bool) {
+	lines, ok := reader.files[path]
+	if !ok {
+		// Read and cache file contents.
+		lines = []string{""} // Skip 0th line
+		f, err := openSourceFile(path, reader.searchPath)
+		if err != nil {
+			reader.errors[path] = err
```

Comment:
Cover the error path?

Reply:
I was not sure what caused "reader.errors[path] = err" line under the "err != nil" condition to be covered, which is what the coverage says now. The test case provides the source files so my feeling was it shouldn't get into the error handler here. Maybe the coverage info is off.